src/lib/validation.ts

```diff
@@ -685,6 +708,23 @@ export class MessageBag extends Map<string, Array<string>> {
   }
 }
 
+/**
+ * Returns a MessageBag object from an Axios error.
+ */
+export const getValidationErrorsFromAxiosError = (err: any): MessageBag => {
+  const errorBag = new MessageBag()
+  const validationErrors = err.response?.data?.data?.validation?.errors
+
+  if (validationErrors) {
+    validationErrors.forEach((validationError: any) => {
+      const errorMessages = errorBag.get(validationError.parameter) || []
+      errorMessages.push(validationError.message)
+      errorBag.set(validationError.parameter, errorMessages)
+    })
+  }
+  return errorBag
+}
```

Comment:
Shouldn't you check for a ValidationError instance? 
Please don't tell me the format is different for 2FA requests :pray:

Reply:
No. This function is the externalization of [these lines](https://github.com/NethServer/nethsecurity-ui/pull/153/files/1f6fd80c23ec3956677226cbfee2b0130a9350df#diff-2e3fe53b79c06a0f2eada22f9bc6ea7eabdbffaa67bc8760a94d930b68959586L40-L49). This way validation errors are now detected and handled also from API calls, not only ubus